pkg/cri/resource-manager/policy/builtin/podpools/podpools-policy.go

```diff
@@ -178,7 +178,7 @@ func (p *podpools) Description() string {
 // Start prepares this policy for accepting allocation/release requests.
 func (p *podpools) Start(add []cache.Container, del []cache.Container) error {
 	log.Info("%s policy started", PolicyName)
-	return p.Sync(add, del)
+	return p.Sync(p.cch.GetContainers(), nil)
```

Comment:
Is that `nil` as deleted containers really correct ? Isn't there another corner case (besides shutdown, change config, start up again, which this attempts to fix IIUC) where some containers are gone while cri-resmgr/the policy is down/away ? And if there is such a case, won't this hack cause those container deletions to go unnoticed and have lingering resource allocations after this change ? It would feel more correct to try freeing everything in del and then allocate everything for all known containers. And that should be the result of calling `p.Sync(p.cch.GetContainers(), del)`, because that's how Sync should work: first free removed containers, then allocate added ones.

Reply:
Thanks @klihub , fixed this issue. Added PR #966 to fix the same issue in the balloons policy.